### ISSUE TITLE
Support "Create branch" link form Jira

### ISSUE THREAD

**Issue by arieteter** [atlassian/github-for-jira#402]
The link in Jira sends to Bitbucket instead of Github

**Comment by mikhaelsantos** [comment]
Any planning for creating a branch in Github?

**Comment by rachellerathbone** [comment]
Hi @arieteter and @mikhaelsantos. I'm from a team at Atlassian that is currently in the process of migrating the ownership over to us from GitHub. As a part of this, we're currently going through the backlog of issues and prs to see what can still be addressed and what should be closed.

This is something my team is actively working on. Due to several factors while we are in the process of the migration, we won't be able to merge our changes for 'create branch', but plan to deploy as soon as we have full ownership of the app. Stay tuned 😄

**Comment by rbehle** [comment]
Any update on this issue?

**Comment by mboudreau** [comment]
@rbehle Very sorry for the wait.  It's been quite hectic here while we try to migrate the app to full Atlassian ownership.  This sadly fell through the cracks in that process.  We are very close to completing the migration work and we can continue the work done for this.  My estimate would be that it would be done in the next few weeks - migration first, then this would be a quick win as we already have most of the work already done.

Thanks for your patience.

**Comment by lucduong** [comment]
@mboudreau Appreciated

**Comment by ricardf** [comment]
Really would appreciate this feature, create the branch from jira and get the branch <> issue linked! any news regarding this?

**Comment by tykim-hd** [comment]
@mboudreau Any ETA on this issue? Developers keep asking about this feature.   ;)

**Comment by rachellerathbone** [comment]
Hey @tykim-hd, @ricardf, @lucduong, @rbehle, @arieteter, and @mikhaelsantos. Apologies for the slow movement on this. Post-migration the team has had to focus on some reliability work but we are now able to start picking up feature work. The stream of work for this hasn't been assigned to anyone on the team yet but I know it is one we want to start focusing on soon. 

Thanks for your patience. I will update here again when I have more information.

**Comment by devnetspot** [comment]
Any update on this matter please ? we really need this feature ! I appreciate if you could provide an estimate date for delivering this feature ? we are all awaiting for create branch within Jira for Github repo  !

**Comment by mboudreau** [comment]
Hi, sorry for the radio silence.  We've been hard at work trying to improve reliability and scalability for our customers and haven't had the time to concentrate on the create branch functionality.  We are having internal talks about it - we had a design review for it just a few days ago - to try to consolidate the experience between the Bitbucket and Github create branch.

I think this might be a long term approach though and doesn't solve your current issue.  We might look into create a short term solution to get the functionality working first before improving upon it further.  I'll bring this concern with the team to see where the prioritization should be.

Cheers.

**Comment by tony-stecca-d2l** [comment]
mboudreau, thanks for the update. It's encouraging to hear that it's being discussed.  We are committed to Jira Software and look forward to this feature.

**Comment by tony-stecca-d2l** [comment]
I notice that the Create Branch feature of the GitHub for Jira integration has changed.  It now provides you with a git command to generate a branch on your machine. It's a nice step forward (see image). @mboudreau is this the short term solution you were proposing?  Still keen to see this integration create a branch in the GitHub repo.

![image](https://user-images.githubusercontent.com/88551204/148269042-52de4005-d14f-401c-bfe7-9cd72430cd48.png)

**Comment by thombergs** [comment]
Hey @tony-stecca-d2l,

this is a general improvement to the "Create Branch" functionality that was made independently from our own plans. 

We still have the "create branch" support for GitHub in our scopes. We want to explore a solution this quarter and implement it next quarter. That's the plan, at least :). Sorry, I can't promise more right now :(.

**Comment by PaulDokken** [comment]
Any status on this?  According to this [Atlassian video](https://www.youtube.com/watch?v=e5yr7nIqTc8) it looks like it should be working but I can't get it to work.

**Comment by ricardf** [comment]
> Any status on this?  According to this [Atlassian video](https://www.youtube.com/watch?v=e5yr7nIqTc8) it looks like it should be working but I can't get it to work.

The part or the video is working. What is missing is the Create GitHub button (we only have gitlab and bitbucket options).

**Comment by ecorichard** [comment]
Would be great to see when will this be available

**Comment by packplusplus** [comment]
Love the create branch shortcut. We would like to customize the branch name suggested to match our naming strategy.  What's the right way to give feedback on adding feature? 

Also added a +1 to https://jira.atlassian.com/browse/JSWCLOUD-22152

**Comment by ehaselwanter** [comment]
Sad to see that this is still not implemented :-(

**Comment by tekNorah** [comment]
> Love the create branch shortcut. We would like to customize the branch name suggested to match our naming strategy. What's the right way to give feedback on adding feature?
> 
> Also added a +1 to https://jira.atlassian.com/browse/JSWCLOUD-22152

Fyi

> Love the create branch shortcut. We would like to customize the branch name suggested to match our naming strategy. What's the right way to give feedback on adding feature?
> 
> Also added a +1 to https://jira.atlassian.com/browse/JSWCLOUD-22152

FYI - instead of +1, please use the vote option

**Comment by rachellerathbone** [comment]
Hey all. Thanks for your patience on this one. We're officially kicking off work next week to support Create Branch for GitHub.

**Comment by rachellerathbone** [comment]
Hey again. Just a quick update. The team working on this is very close to wrapping up and then will start rolling this out in prod. Will update again when that's done.

**Comment by timothywlewis** [comment]
@rachellerathbone glad to hear this! Will it take into account which repo "should" be associated with a project, so that the branch is created in the correct repo? #585

**Comment by rachellerathbone** [comment]
Hey @timothywlewis. Yes 😄 When you click the create branch link, you'll be navigated to a screen where you will select the org and repo you want to create the branch in. Keep in mind, you'll only see the orgs you've connected to via our integration, and the repos you've given permissions to.

We're keeping our fingers crossed this will start rolling out tomorrow. Waiting on a couple of things on the Jira side of things but we're very close 🙏

**Comment by aminzar-access** [comment]
Hi @rachellerathbone, any updates on this? Thanks :)

**Comment by KyeRussell** [comment]
I think a couple of weeks ago I saw this flash up for a moment or two on my Jira instance. It disappeared after an hour or so. Assuming this was intentional and this still hasn't rolled out?

**Comment by engineerakki** [comment]
Hi @rachellerathbone 

Is this feature rolled out yet ? Because we cannot see it in
We would love to know updates / ETA for this feature.

**Comment by jsdse** [comment]
Cool, that looks great! This is for the cloud version right? is there any plan for the Data Center version of Jira?

**Comment by aminzar-access** [comment]
@rachellerathbone I just checked, and it is working but it is not using the format I have set!

**Comment by foundingnimo** [comment]
It seems like we should be able to configure the template for the branch name - but it isn't being honoured, and always uses <issue-key>-<issue-summary-short>
Doesn't matter what template I try to use (with the collection of possible fields) - the url that 'create branch' navigates to is always 
`https://github.atlassian.com/github/create-branch?issueKey=xxx&issueSummary=yyy&....` 
which results in the the branch name being xxx-yyy.

The generated command (to run locally) seems to be correct :)

**Comment by aminzar-access** [comment]
@rachellerathbone is there any plans to resolve this issue? Thanks

**Comment by atrigueiro** [comment]
Hi all, this feature has been rolled out a few months ago. I'm closing this issue but please feel free to raise a new one if you encounter any issues when creating a branch.

Cheers :)
